internal/gps/source_cache.go

```diff
@@ -59,6 +59,9 @@ type singleSourceCache interface {
 	// If the input is a revision and multiple UnpairedVersions are associated
 	// with it, whatever happens to be the first is returned.
 	toUnpaired(v Version) (UnpairedVersion, bool)
+
+	// close closes the cache and any backing resources.
+	close() error
```

Comment:
Given that the [doc](https://golang.org/pkg/io/#Closer) is so open ended for this particular interface, I'm not sure that it buys us much. I also hesitate to export a method before it needs to be. I don't really have strong feelings either way though.